### PR TITLE
[Php80] Fix str_ends_with() when needle is longer than haystack

### PR DESCRIPTION
**Pull request by SpacePossum**
Little bug, found in https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/6323

https://3v4l.org/oHUKH 

```php
<?php
error_reporting(E_ALL);
$needle = '[]';
echo \substr_compare('', $needle, -\strlen($needle));
```

show issues for

```
Output for 5.3.0 - 5.3.29, 5.4.0 - 5.4.45, 5.5.0 - 5.5.38, 5.6.0 - 5.6.40, 7.0.0 - 7.0.33, 7.1.0 - 7.1.33, 7.2.0 - 7.2.17, 7.3.0 - 7.3.4
Warning: substr_compare(): The start position cannot exceed initial string length in /in/oHUKH on line 4
```

Looking at the test matrix I assume the tests will run latest 7.2 which is not effected, 
~so I naively changed it here in this draft to see if my test indeeds fails on the CI (I don't have an easy 7.2.17 or 7.1.0 env. at hand at the moment)~ @ see https://github.com/shivammathur/setup-php/issues/570

### CODE REVIEW

File: tests/Php80/Php80Test.php [hunk 1]
```diff
@@ -192,6 +192,8 @@ public function testStrEndsWith()
         $testEmoji = '🙌🎉✨🚀'; // 0xf0 0x9f 0x99 0x8c 0xf0 0x9f 0x8e 0x89 0xe2 0x9c 0xa8 0xf0 0x9f 0x9a 0x80
         $this->assertTrue(str_ends_with($testEmoji, '🚀')); // 0xf0 0x9f 0x9a 0x80
         $this->assertFalse(str_ends_with($testEmoji, '✨')); // 0xe2 0x9c 0xa8
+
+        $this->assertFalse(str_ends_with('', '[]')); // 0xe2 0x9c 0xa8
```
Comment (jderusse) [inline tests/Php80/Php80Test.php:196]:
```suggestion
        $this->assertFalse(str_ends_with('', '[]'));
```

Comment (jderusse) [inline tests/Php80/Php80Test.php:196]:
```suggestion
        $this->assertFalse(str_ends_with('', '[]'));
```

**Review by derrabus — APPROVED** [review]

**Review by keradus — APPROVED** [review]
🍻

**Review by IonBazan — APPROVED** [review]
Nice finding. Cheers!

**Comment by nicolas-grekas** [comment]
Thank you @SpacePossum.

**Comment by SpacePossum** [comment]
Thank you all :)

**Pull request merged**